### DISCUSSION TITLE
MapLibre GL JS performance with raster tiles & RMB context menu

### DISCUSSION THREAD

**Discussion by rene78** [maplibre/maplibre-gl-js#2104] [Q&A] [answered]
Hi everyone!

I thought about switching from _Leaflet_ JS to _MapLibre GL JS_ for the project [_Latest Changes_](https://github.com/rene78/latest-changes). 2 questions came to my mind while checking out some examples:

1. The only way of getting vector map tiles for free without self hosting seems to be the API of Maptiler. For this an API key is needed. Thus, if someone else decides to fork the repository he/she will have to sign up to Maptiler as well in order to display the map. Suboptimal in my opinion. That's why I thought of using the raster tiles from "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}". The question would be: **Is MapLibre still superior to Leaflet when it comes to displaying a lot of GeoJSON elements, even if the base map consists of raster tiles?** (I would think yes?)
2. On Leaflet there is a plugin to show right mouse button context menus ([leaflet.contextmenu](https://github.com/aratcliffe/Leaflet.contextmenu)). Is there something similar with MapLibre?

Thanks very much in advance!
Rene

**Comment by HarelM** [comment, marked as answer]
1. Layer are independent and so are the sources, so yes, maplibre displays geojson fast. You can use the demotiles repo for vector tiles, but it's only up to zoom 7, depending on your needs.
2. You'll need to write something, it should be very simple though, I'm not aware of a plugin and it depends on what you want to put in that menu. You can use a popup that comes built it.
